theme watch - "Runtime Error: invalid memory address or nil pointer dereference" after many updates in v0.5.1

I upgraded to v0.5.1 yesterday, and when saving files to be updated by `theme watch`, a `panic: runtime error: invalid memory address or nil pointer dereference` error is thrown.

The seems to be occurring after saving the file ~12 times, which is then followed by `theme watch` crashing.

Here is the full error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6a018]

goroutine 97 [running]:
panic(0x36dd20, 0xc4200120e0)
        /usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1a1
github.com/Shopify/themekit/kit.(*httpClient).AssetAction(0xc4200f9200, 0x2, 0xc4203b8050, 0xe, 0xc4204d4000, 0xeafe, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/kit/http_client.go:112 +0x1e8
github.com/Shopify/themekit/kit.ThemeClient.Perform(0xc4203dd520, 0x20, 0xc4203d5958, 0x4, 0xc4203309c0, 0x27, 0xc420014084, 0x2e, 0xc4202be080, 0x3, ...)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/kit/theme_client.go:156 +0x311
github.com/Shopify/themekit/cmd.handleWatchEvent(0xc4203dd520, 0x20, 0xc4203d5958, 0x4, 0xc4203309c0, 0x27, 0xc420014084, 0x2e, 0xc4202be080, 0x3, ...)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/cmd/watch.go:67 +0x364
github.com/Shopify/themekit/kit.handleEvent(0xc4203f23c0, 0xc420315940, 0x3d, 0xc400000002)
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/kit/file_watcher.go:140 +0x209
created by github.com/Shopify/themekit/kit.callbackEvents
        /Users/timanema/workspace/go/src/github.com/Shopify/themekit/kit/file_watcher.go:94 +0x14e
```

Additional details:

- Go v1.7.1 (darwin/amd64)
- macOS Sierra - v10.12 (16A323)
- File is a JS file in the `assets` dir

Let me know if there is any additional info I can provide, and I'll share anything else I come across.

--- a/kit/http_client.go
+++ b/kit/http_client.go
@@ -105,9 +105,12 @@ func (client *httpClient) GetTheme(themeID int64) (*ShopifyResponse, Error) {
 }
 
 func (client *httpClient) AssetAction(event EventType, asset Asset) (*ShopifyResponse, Error) {
-	resp, _ := client.sendJSON(assetRequest, event, client.AssetPath(), map[string]interface{}{
+	resp, err := client.sendJSON(assetRequest, event, client.AssetPath(), map[string]interface{}{
 		"asset": asset,
 	})
+	if resp == nil {
+		return resp, err
+	}
 	// If there were any errors the asset is nil so lets set it and reformat errors
 	resp.Asset = asset
 	return resp, resp.Error()

--- a/kit/http_client.go
+++ b/kit/http_client.go
@@ -105,21 +105,18 @@ func (client *httpClient) GetTheme(themeID int64) (*ShopifyResponse, Error) {
 }
 
 func (client *httpClient) AssetAction(event EventType, asset Asset) (*ShopifyResponse, Error) {
-	resp, err := client.sendJSON(assetRequest, event, client.AssetPath(), map[string]interface{}{
+	resp, _ := client.sendJSON(assetRequest, event, client.AssetPath(), map[string]interface{}{
 		"asset": asset,
 	})
-	if resp == nil {
-		return resp, err
-	}
 	// If there were any errors the asset is nil so lets set it and reformat errors
 	resp.Asset = asset
 	return resp, resp.Error()
 }
 
-func (client *httpClient) newRequest(event EventType, urlStr string, body io.Reader) (*http.Request, Error) {
+func (client *httpClient) newRequest(event EventType, urlStr string, body io.Reader) (*http.Request, error) {
 	req, err := http.NewRequest(event.toMethod(), urlStr, body)
 	if err != nil {
-		return nil, kitError{err}
+		return nil, err
 	}
 
 	req.Header.Add("X-Shopify-Access-Token", client.config.Password)
@@ -133,7 +130,7 @@ func (client *httpClient) newRequest(event EventType, urlStr string, body io.Rea
 func (client *httpClient) sendJSON(rtype requestType, event EventType, urlStr string, body map[string]interface{}) (*ShopifyResponse, Error) {
 	data, err := json.Marshal(body)
 	if err != nil {
-		return nil, kitError{err}
+		return newShopifyResponse(rtype, event, nil, err)
 	}
 	return client.sendRequest(rtype, event, urlStr, bytes.NewBuffer(data))
 }
@@ -141,7 +138,7 @@ func (client *httpClient) sendJSON(rtype requestType, event EventType, urlStr st
 func (client *httpClient) sendRequest(rtype requestType, event EventType, urlStr string, body io.Reader) (*ShopifyResponse, Error) {
 	req, err := client.newRequest(event, urlStr, body)
 	if err != nil {
-		return nil, kitError{err}
+		return newShopifyResponse(rtype, event, nil, err)
 	}
 	apiLimit.Wait()
 	resp, respErr := client.client.Do(req)

--- a/kit/shopify_response.go
+++ b/kit/shopify_response.go
@@ -27,7 +27,10 @@ type ShopifyResponse struct {
 
 func newShopifyResponse(rtype requestType, event EventType, resp *http.Response, err error) (*ShopifyResponse, Error) {
 	if resp == nil || err != nil {
-		return nil, kitError{err}
+		return &ShopifyResponse{
+			Type:      rtype,
+			EventType: event,
+		}, kitError{err}
 	}
 	defer resp.Body.Close()
 
@@ -41,7 +44,7 @@ func newShopifyResponse(rtype requestType, event EventType, resp *http.Response,
 
 	bytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return nil, kitError{err}
+		return newResponse, kitError{err}
 	}
 
 	err = json.Unmarshal(bytes, &newResponse)

--- a/kit/shopify_response_test.go
+++ b/kit/shopify_response_test.go
@@ -18,7 +18,7 @@ func (suite *ShopifyResponseTestSuite) TestRequestError() {
 	errorMessage := "something went wrong"
 	badErr := fmt.Errorf(errorMessage)
 	resp, err := newShopifyResponse(themeRequest, Create, &http.Response{}, badErr)
-	assert.Nil(suite.T(), resp)
+	assert.NotNil(suite.T(), resp)
 	if assert.NotNil(suite.T(), err) {
 		assert.Equal(suite.T(), errorMessage, err.Error())
 	}
@@ -31,7 +31,7 @@ func (suite *ShopifyResponseTestSuite) TestNoBody() {
 	}
 	mock.Body.Close()
 	resp, err := newShopifyResponse(themeRequest, Create, mock, nil)
-	assert.Nil(suite.T(), resp)
+	assert.NotNil(suite.T(), resp)
 	assert.NotNil(suite.T(), err)
 }
 

